xds/internal/balancer/edsbalancer/eds.go

```diff
@@ -142,10 +140,9 @@ func (x *edsBalancer) run() {
 		case <-x.ctx.Done():
 			if x.client != nil {
 				x.client.close()
```

Comment:
Should we make the `client.Close` handle `nil`.

Reply:
`client` already handles multiple `close` calls, though it will never be called the second time, because this goroutine is the only caller.
I removed the nil check and setting nil. It's never nil.